runtime/src/iree/hal/drivers/vulkan/native_executable.cc

```diff
@@ -27,187 +27,179 @@
-    stage_create_info->flags = 0;
-    stage_create_info->stage = VK_SHADER_STAGE_COMPUTE_BIT;
-    uint32_t shader_module_index =
-        flatbuffers_uint32_vec_at(shader_module_indices_vec, entry_ordinal);
-    // We have verified that shader_module_index is within the range.
-    stage_create_info->module = shader_modules[shader_module_index];
-    stage_create_info->pName =
-        flatbuffers_string_vec_at(entry_points_vec, entry_ordinal);
-    stage_create_info->pSpecializationInfo = &spec_info;
-
-    // If subgroup size is not 0, request the said subgroup size via
-    // VK_EXT_subgroup_size_control (promoted to core since v1.3).
-    stage_create_info->pNext = NULL;
-    if (subgroup_sizes_vec) {
-      if (uint32_t subgroup_size =
-              flatbuffers_uint32_vec_at(subgroup_sizes_vec, entry_ordinal)) {
-        subgroup_control_entries[entry_ordinal].sType =
-            VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_REQUIRED_SUBGROUP_SIZE_CREATE_INFO;
-        subgroup_control_entries[entry_ordinal].pNext = NULL;
-        subgroup_control_entries[entry_ordinal].requiredSubgroupSize =
-            subgroup_size;
-        stage_create_info->pNext = &subgroup_control_entries[entry_ordinal];
+    iree_hal_vulkan_DescriptorSetLayoutDef_table_t set_layout_def =
+        iree_hal_vulkan_DescriptorSetLayoutDef_vec_at(
+            descriptor_set_layouts_vec, descriptor_set_layout_ordinal);
+    iree_hal_vulkan_DescriptorSetLayoutBindingDef_vec_t bindings_vec =
+        iree_hal_vulkan_DescriptorSetLayoutDef_bindings_get(set_layout_def);
+    for (iree_host_size_t j = 0;
+         j <
+         iree_hal_vulkan_DescriptorSetLayoutBindingDef_vec_len(bindings_vec);
+         ++j) {
+      iree_hal_vulkan_DescriptorSetLayoutBindingDef_table_t binding_def =
+          iree_hal_vulkan_DescriptorSetLayoutBindingDef_vec_at(bindings_vec, j);
+      uint32_t descriptor_count =
+          iree_hal_vulkan_DescriptorSetLayoutBindingDef_descriptor_count_get(
+              binding_def);
+      iree_hal_vulkan_VkDescriptorType_enum_t type =
+          iree_hal_vulkan_DescriptorSetLayoutBindingDef_descriptor_type_get(
+              binding_def);
+      uint32_t stage_flags =
+          iree_hal_vulkan_DescriptorSetLayoutBindingDef_stage_flags_get(
+              binding_def);
+      switch (type) {
+        case iree_hal_vulkan_VkDescriptorType_UNIFORM_BUFFER:
+          uniform_descriptor_count += descriptor_count;
+          break;
+        case iree_hal_vulkan_VkDescriptorType_STORAGE_BUFFER:
+          storage_descriptor_count += descriptor_count;
+          break;
+        default:
+          return iree_make_status(IREE_STATUS_INVALID_ARGUMENT,
+                                  "pipeline layout set[%" PRIhsz
+                                  "] binding[%" PRIhsz
+                                  "] has "
+                                  "an unsupported descriptor_type",
+                                  i, j);
```

Comment:
nit: I think these lines can be condensed?
```suggestion
          return iree_make_status(IREE_STATUS_INVALID_ARGUMENT,
                                  "pipeline layout set[%" PRIhsz
                                  "] binding[%" PRIhsz
                                  "] has an unsupported descriptor_type",
                                  i, j);
```